### PR TITLE
#RI-3768 - Display the "Scan more" button in the tree view

### PR DESCRIPTION
**Pull request by zalenskiSofteq**
#RI-3768 - Display the "Scan more" button in the tree view

### CODE REVIEW

File: redisinsight/api/test/api/redisearch/POST-databases-id-redisearch-search.test.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ const responseSchema = Joi.object({
   cursor: Joi.number().integer().required(),
   scanned: Joi.number().integer().required(),
   total: Joi.number().integer().required(),
+  maxResults: Joi.number().integer().allow(null).required().allow(null),
```
Comment (arthosofteq) [inline redisinsight/api/test/api/redisearch/POST-databases-id-redisearch-search.test.ts:37]: there are 2 .allow(null) rules

File: redisinsight/api/src/modules/browser/dto/keys.dto.ts [hunk 2]
```diff
@@ -319,4 +319,13 @@ export class GetKeysWithDetailsResponse {
     description: 'Node port. In case when we are working with cluster',
   })
   port?: number;
+
+  @ApiPropertyOptional({
+    type: Number,
+    description:
+      'The maximum number of results.'
+      + ' For RediSearch this number is a value from "FT.CONFIG GET maxsearchresults" command.'
+      + ' If FT.CONFIG unavailable the number is 10.000 by default for RediSearch',
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/browser/dto/keys.dto.ts:328]: `If FT.CONFIG unavailable the number is 10.000 by default for RediSearch`
Is this this true?

File: redisinsight/api/src/modules/browser/services/redisearch/redisearch.service.ts [hunk 3]
```diff
@@ -147,16 +150,33 @@ export class RedisearchService {
...
+        maxResults,
       });
-    } catch (e) {
-      this.logger.error('Failed to search keys using redisearch index', e);
-
-      throw catchAclError(e);
+    } catch (error) {
+      this.logger.error('Failed to search keys using redisearch index', error);
+      if (error.message?.includes(RedisErrorCodes.RedisearchLimit)) {
+        throw new BadRequestException(ERROR_MESSAGES.INCREASE_MINIMUM_LIMIT(dto.limit?.toLocaleString?.()));
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/browser/services/redisearch/redisearch.service.ts:177]: `ERROR_MESSAGES.INCREASE_MINIMUM_LIMIT(dto.limit?.toLocaleString?.())`
May be `dto.offset` or `dto.limit + dto.offset`? ideally limit will never change across "show more" requests (e.g. always 10 000)

File: redisinsight/api/src/constants/error-messages.ts [hunk 4]
```diff
@@ -52,4 +52,5 @@ export default {
   REDIS_MODULE_IS_REQUIRED: (module: string) => `Required ${module} module is not loaded.`,
   APP_SETTINGS_NOT_FOUND: () => 'Could not find application settings.',
   SERVER_INFO_NOT_FOUND: () => 'Could not find server info.',
+  INCREASE_MINIMUM_LIMIT: (count: string) => `Use a minimum of ${count} as the MAXSEARCHRESULTS.`,
```
Comment (arthosofteq) [inline redisinsight/api/src/constants/error-messages.ts:55]: Maybe I'm wrong, but I don't agree with such error message. What does it represents? What we are trying to "avoid" by implementing this entire feature?

**Review by arthosofteq — CHANGES_REQUESTED** [review]
The code looks good. Great job! 
However I don't understand the feature. Could you explain me, please?
I tough the we must show something like `"It is impossible to search for {dto.offset}+ entries since MAXSEARCHRESULTS = {FT.MAXSEARCHRESULTS} limit exceeded... etc..."`. and do not show "Show more" button if offset >= FT.MAXSEARCHRESULTS...

File: redisinsight/api/test/api/redisearch/POST-databases-id-redisearch-search.test.ts (hunk 1 above)
Comment (zalenskiSofteq) [reply redisinsight/api/test/api/redisearch/POST-databases-id-redisearch-search.test.ts]: removed

File: redisinsight/api/src/modules/browser/dto/keys.dto.ts (hunk 2 above)
Comment (zalenskiSofteq) [reply redisinsight/api/src/modules/browser/dto/keys.dto.ts]: This is for Cloud databases. Other databases may have differences.
Removed.

File: redisinsight/api/src/modules/browser/services/redisearch/redisearch.service.ts (hunk 3 above)
Comment (zalenskiSofteq) [reply redisinsight/api/src/modules/browser/services/redisearch/redisearch.service.ts:177]: Only config for limit throws an error. For the Browser view limit is 500.

File: redisinsight/api/src/constants/error-messages.ts (hunk 4 above)
Comment (zalenskiSofteq) [reply redisinsight/api/src/constants/error-messages.ts]: Viktor sent a new message.

**Review by arthosofteq — APPROVED** [review]
Still concerned about error message. 
Everything else looks good. Well done!

**Pull request merged**